$MirOS: ports/www/php/patches/patch-ext_session_session_c,v 1.1.7.1 2005/03/18 16:00:00 tg Exp $
--- ext/session/session.c.orig	Thu Mar 24 00:17:31 2005
+++ ext/session/session.c	Wed May  4 14:38:07 2005
@@ -653,8 +653,8 @@ PHPAPI char *php_session_create_id(PS_CR
 	buf = emalloc(100);
 
 	/* maximum 15+19+19+10 bytes */	
-	sprintf(buf, "%.15s%ld%ld%0.8f", remote_addr ? remote_addr : "", 
-			tv.tv_sec, tv.tv_usec, php_combined_lcg(TSRMLS_C) * 10);
+	sprintf(buf, "%.15s%lld%ld%0.8f", remote_addr ? remote_addr : "", 
+			(long long)tv.tv_sec, tv.tv_usec, php_combined_lcg(TSRMLS_C) * 10);
 
 	switch (PS(hash_func)) {
 	case PS_HASH_FUNC_MD5:
@@ -857,7 +857,7 @@ static void strcpy_gmt(char *ubuf, time_
 	
 	n = sprintf(buf, "%s, %02d %s %d %02d:%02d:%02d GMT", /* SAFE */
 				week_days[tm.tm_wday], tm.tm_mday, 
-				month_names[tm.tm_mon], tm.tm_year + 1900, 
+				month_names[tm.tm_mon], (int)tm.tm_year + 1900, 
 				tm.tm_hour, tm.tm_min, 
 				tm.tm_sec);
 	memcpy(ubuf, buf, n);
